packages/ui-components/src/components/InputGroup/InputGroup.component.tsx

```diff
@@ -43,15 +43,24 @@ const cloneElementWithInheritedProps = (
   parentVariant: VariantTypes,
   parentDisabled: boolean
 ): ReactNode => {
-  if (!isValidElement<OverrideProps>(child)) return child
+  if (!isValidElement<InheritedProps>(child)) return child
 
   const combinedProps = {
-    variant: child.props.variant ?? parentVariant,
-    disabled: child.props.disabled ?? parentDisabled,
+    variant: child.props.variant || parentVariant,
+    disabled: child.props.disabled || parentDisabled,
   }
   return React.cloneElement(child, combinedProps)
 }
 
+const getClassNames = (
+  className: string,
+  variant: VariantTypes,
+  disabled: boolean,
+  additionalClassName: string
+): string => {
+  return `${className} ${className}-${variant} ${disabled ? `${className}-disabled` : ""} ${additionalClassName}`
```

Comment:
One comment regarding using string interpolation to construct class names. This is fine in this case because those are custom class names but it wouldn't work for tailwind class names. So be careful with string interpolation in classnames. Find more info in the tailwind documentation about dynamic classnames: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
Two things here:
1. To stay consistent with other components the className function should only produce the classNames that deal with the styling of the element. The best practise is to specify the semantic name (`className` or `juno-input-group` here) in the jsx/tsx markup. Same with the `className` from props. That way it's easier to see when looking at the markup what the respective elements are and in case of the `className` from props for complex components there are multiple choices which inner element to apply those class names to and it's better if you can see this in the jsx/tsx markup section.
2. We typically write class name strings in multiple lines because this makes it easier to read.
```suggestion
  return `
      ${className}-${variant} 
      ${disabled ? `${className}-disabled` : ""} 
  `
```